### DISCUSSION TITLE
Tasmota used to create a thesis (Computer Engineering)

### DISCUSSION THREAD

**Discussion by FulvioSpelta** [arendst/Tasmota#22549] [Show and tell]
I accidentally found a thesis based on tasmota and I'm posting it here as a compliment for your work!
[tesi.pdf](https://webthesis.biblio.polito.it/31829/1/tesi.pdf)

**Comment by arendst** [comment]
Wow. Any chance to have an english translation?

**Reply by FulvioSpelta** [reply]
I don't think because the work is closed. If you need something ask me and i'll try to give you info or contact the author.
Quick summary
The guy has used tasmota to build a "smart" energy distribution unit that collect info and send them to a control center.
It has been applied to a mini mining farm just to create emphasis about the usefulness of the energy monitor and management.

IMHO is a "usual" use case in this community, it's nice that has been used in thesis for an engineering degree and the focus, that they have clearly underlined, about the value in term of time and economic value that is created by using professional open software.

**Reply by sfromis** [reply]
Of course, machine translation seems to work reasonably; unfortunately, Google translate won't translate the full PDF in a single step as it is well above 10 MB. Still, marking a paragraph and selecting translate, works too - at least in Chrome. 

Two paragraphs from the summary:

*The paper describes the modifications required to the Tasmota firmware to adapt it to specific use cases. This includes user-configurable settings and advanced code changes, emphasizing the use of the MQTT protocol to delegate logic processing to external components. This approach is essential because the processing power of the ESP32 is insufficient for the complex logic operations required in a mining setup.*

*A practical implementation of Tasmota is shown using the BioPDU, a PDU based on the Tasmota firmware, which allows remote management and provides automatic notifications of malfunctions via Telegram.*

**Comment by arendst** [comment]
It even has it's own driver `xnrg_24_biopdu.ino` in our repository.

**Comment by ZioFabry** [comment]
Sorry... I missed this discussion before...
I was the sponsor & relator of the thesis...

here you can see the final result of the project... not ready yet during the thesis discussion...

https://biomine.it/2024/10/building-a-mining-farm/

**Comment by ZioFabry** [comment]
![image](https://github.com/user-attachments/assets/e3cf0528-2b8e-48c8-9cf8-8442a40ded68)

**Comment by ZioFabry** [comment]
![image](https://github.com/user-attachments/assets/842d5f21-3e3e-4021-88f6-391f5753ed13)
![image](https://github.com/user-attachments/assets/67cb3b26-9979-4cbc-aca9-236700260325)
